<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>863</width>
    <height>571</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Service</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="TabWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>841</width>
      <height>551</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="RequestViewsTab">
     <attribute name="title">
      <string>Request Views</string>
     </attribute>
     <widget class="QListWidget" name="OnlineUsersList">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>60</y>
        <width>311</width>
        <height>291</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Online Users</string>
      </property>
     </widget>
     <widget class="QPushButton" name="ViewUserImagesButton">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>370</y>
        <width>231</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>View Selected User's Images</string>
      </property>
     </widget>
     <widget class="QTreeWidget" name="UserImagesTree">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>60</y>
        <width>391</width>
        <height>291</height>
       </rect>
      </property>
      <property name="columnCount">
       <number>4</number>
      </property>
      <attribute name="headerCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="headerDefaultSectionSize">
       <number>75</number>
      </attribute>
      <attribute name="headerHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="headerStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column>
       <property name="text">
        <string notr="true">ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string notr="true">Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Owner</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Views</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>40</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>User's Images</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>380</y>
        <width>121</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Number of Views</string>
      </property>
     </widget>
     <widget class="QPushButton" name="RequestViewsButton">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>430</y>
        <width>261</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Request Views' Update From Owner</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>60</y>
        <width>31</width>
        <height>291</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="NumberOfViewsSpinBox">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>370</y>
        <width>61</width>
        <height>41</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Pending Requests</string>
     </attribute>
     <widget class="QTreeWidget" name="OtherUpdateRequestsTreeWidget">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>80</y>
        <width>341</width>
        <height>311</height>
       </rect>
      </property>
      <column>
       <property name="text">
        <string notr="true">ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Username</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Requested Views</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>60</y>
        <width>231</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Other Users' Update Requests</string>
      </property>
     </widget>
     <widget class="QPushButton" name="AcceptRequestButton">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>400</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Accept Request</string>
      </property>
     </widget>
     <widget class="QPushButton" name="DenyRequestButton">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>400</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Deny Request</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="MyImages">
     <attribute name="title">
      <string>My Images</string>
     </attribute>
     <widget class="QTreeWidget" name="OtherImagesTree">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>300</y>
        <width>431</width>
        <height>161</height>
       </rect>
      </property>
      <attribute name="headerDefaultSectionSize">
       <number>100</number>
      </attribute>
      <column>
       <property name="text">
        <string notr="true">ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Owner</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Number of Views</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>81</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>My Images</string>
      </property>
     </widget>
     <widget class="QPushButton" name="UploadNewImageButton">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>210</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Upload New Image</string>
      </property>
     </widget>
     <widget class="QComboBox" name="AllowedPeersComboBox">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>50</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>80</y>
        <width>241</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Remaining views for selected user: </string>
      </property>
     </widget>
     <widget class="QTreeWidget" name="MyImagesTree">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>241</width>
        <height>201</height>
       </rect>
      </property>
      <property name="columnCount">
       <number>2</number>
      </property>
      <attribute name="headerDefaultSectionSize">
       <number>10</number>
      </attribute>
      <attribute name="headerMinimumSectionSize">
       <number>57</number>
      </attribute>
      <column>
       <property name="text">
        <string>ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>30</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Image Thumbnail</string>
      </property>
     </widget>
     <widget class="QLabel" name="ImageThumbnail">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>60</y>
        <width>181</width>
        <height>131</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="MyImageViewQuota">
      <property name="geometry">
       <rect>
        <x>720</x>
        <y>80</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>30</y>
        <width>131</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Authorized Users</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>140</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Online Users</string>
      </property>
     </widget>
     <widget class="QPushButton" name="DeleteUserButton">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>50</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete User</string>
      </property>
     </widget>
     <widget class="QComboBox" name="OnlinePeersComboBox">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>160</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>140</y>
        <width>141</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>New Viewing Quota</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="NewViewingQuotaSpinBox">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>160</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="AddOrUpdateButton">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>210</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Add or Update User</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>280</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Others' Images</string>
      </property>
     </widget>
     <widget class="QPushButton" name="ViewImageButton">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>470</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>View Image</string>
      </property>
     </widget>
     <widget class="QLabel" name="RemainingViews">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>470</y>
        <width>67</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="RemoteImage">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>300</y>
        <width>351</width>
        <height>201</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="ForceUpdateButton">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>470</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Refresh Now</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
